new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -0,0 +1,557 @@
+          .isTrue();
+
+      //TO Do: Create docker registry secret
+    }
+  }
+
+  private void createSecretForDomain(String secretName, String username, String password) {
+    Map<String, String> secretMap = new HashMap();
+    secretMap.put("username", username);
+    secretMap.put("password", password);
+
+    try {
+      assertThat(createSecret(new V1Secret()
+          .metadata(new V1ObjectMeta()
+              .name(secretName)
+              .namespace(domainNamespace))
+          .stringData(secretMap)))
+          .as("Test createSecret returns true")
+          .withFailMessage("createSecret failed")
+          .isTrue();
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "createSecret failed with ", e);
+      assertThat(e)
+          .as("Test that createSecret does not throw an exception")
+          .withFailMessage(String.format("Create secret %s failed while creating secret "
+              + "for admin credentials", secretName))
+          .isNotInstanceOf(ApiException.class);
+    }
+  }
+
+  private void createDomain(Domain domain) {
+    boolean result = false;
+    try {
+      result = createDomainCustomResource(domain);
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "createDomainCustomResource failed with ", e);
+      assertThat(e)
+          .as("Test that createDomainCustomResource does not throw an exception")
+          .withFailMessage(String.format(
+              "Could not create domain custom resource for domainUID %s in namespace %s",
+              domainUID, domainNamespace))
+          .isNotInstanceOf(ApiException.class);
+    }
+    assertThat(result)
+        .as("Test createDomainCustomResource returns true")
+        .withFailMessage(String.format(
+            "Create domain custom resource failed for domainUID %s in namespace %s",
+            domainUID, domainNamespace))
+        .isTrue();
+
+  }
+
+  private void checkPodCreated(String podName) {
+    try {
+      waitForCondition(podExists(podName, domainUID, domainNamespace));
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "podExists failed with ", e);
+      assertThat(e)
+          .as("Test that podExists does not throw an exception")
+          .withFailMessage(String.format(
+              "pod %s doesn't exist in namespace %s", podName, domainNamespace))
+          .isNotInstanceOf(ApiException.class);
+    }
+  }
+
+  private void checkPodRunning(String podName) {
+    try {
+      waitForCondition(podReady(podName, domainUID, domainNamespace));
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "podReady failed with ", e);
+      assertThat(e)
+          .as("Test that podReady does not throw an exception")
+          .withFailMessage(String.format(
+              "pod %s is not ready in namespace %s", podName, domainNamespace))
+          .isNotInstanceOf(ApiException.class);
+    }
+  }
+
+  private void checkServiceCreated(String serviceName) {
```

Comment:
This method can be moved to a utility class so that every IT class can use it.

Reply:
sure, we can discuss about this in the meeting today.